src/Application/StatementListTranslator.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseFacetedSearch\Application;
+
+use Wikibase\DataModel\Entity\ItemId;
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\DataModel\Statement\StatementList;
+
+class StatementListTranslator {
+
+	public function __construct(
+		private readonly StatementTranslator $statementTranslator,
+		private readonly InstanceTypeExtractor $instanceTypeExtractor,
+		private readonly Config $config
+	) {
+	}
+
+	/**
+	 * @return array<string, mixed>
+	 */
+	public function translateStatements( StatementList $statements ): array {
+		$instanceTypeId = $this->instanceTypeExtractor->getInstanceTypeId( $statements );
+
+		if ( $instanceTypeId === null ) {
+			return [];
+		}
+
+		$propertyIds = $this->getInstanceTypePropertyIds( $instanceTypeId );
+
+		$values = [];
+
+		foreach ( $propertyIds as $propertyId ) {
+			$values[$this->getFieldName( $propertyId )] = $this->translatePropertyStatements( $statements, $propertyId );
+		}
+
+		return $values;
+	}
+
+	/**
+	 * @return PropertyId[]
+	 */
+	private function getInstanceTypePropertyIds( ItemId $instanceTypeId ): array {
+		return array_values(
+			array_map(
+				fn( FacetConfig $config ) => $config->propertyId,
+				$this->config->getFacetConfigForInstanceType( $instanceTypeId )
+			)
+		);
+	}
+
+	private function translatePropertyStatements( StatementList $statements, PropertyId $propertyId ): array {
+		$values = [];
+		$propertyStatements = $statements->getByPropertyId( $propertyId )->getBestStatements()->toArray();
```

Comment:
I'm just using `getBestStatements()` (which returns `RANK_PREFERRED` _or else_ `RANK_NORMAL`).
In https://github.com/ProfessionalWiki/SemanticWikibase/blob/master/src/Translation/StatementListTranslator.php#L26 it's also calling `getByRank()` afterwards, which seems to be redundant, since it would just return the same already filtered list.
However, if the idea is to get `RANK_PREFERRED` _and_ `RANK_NORMAL`, then we should remove `getBestStatements()` and only use `getByRank()`.

Reply:
Just `getBestStatements` is correct